openmeter/ent/schema/subscription_patch.go

```diff
@@ -0,0 +1,212 @@
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/dialect"
+	"entgo.io/ent/dialect/entsql"
+	"entgo.io/ent/schema"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+)
+
+type SubscriptionPatch struct {
+	ent.Schema
+}
+
+func (SubscriptionPatch) Annotations() []schema.Annotation {
+	return []schema.Annotation{
+		entsql.Annotation{Table: "subscription_patches"},
+	}
+}
+
+func (SubscriptionPatch) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		entutils.IDMixin{},
+		entutils.NamespaceMixin{},
+		entutils.TimeMixin{},
+		entutils.MetadataAnnotationsMixin{},
+	}
+}
+
+func (SubscriptionPatch) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("subscription_id").NotEmpty().Immutable(),
+		field.Time("applied_at").Immutable(),
+		field.Int("batch_index").Immutable(),
+		field.String("operation").NotEmpty().Immutable(),
```

Comment:
should this be a type of Go enum like?
`field.String("type").GoType(appentitybase.AppType("")).Immutable(),`